src/libAtomVM/externalterm.c

```diff
@@ -355,6 +400,19 @@ static term parse_external_terms(const uint8_t *external_term_buf, int *eterm_si
             return term_make_maybe_boxed_int64(ctx, value);
         }
 
+        case SMALL_BIG_EXT: {
+            uint8_t num_bytes = external_term_buf[1];
+            uint8_t sign = external_term_buf[2];
+            avm_uint64_t unsigned_value = read_bytes(external_term_buf + 3, num_bytes);
+            avm_int64_t value = 0;
+            if (sign != 0x00) {
+                value = ((avm_int64_t) unsigned_value) * -1;
```

Comment:
here again, might be an undefined: when unsigned_value is `-INT_MIN64`, `((avm_int64_t) unsigned_value) * -1` will be undefined because it is casted to `avm_int64_t` before applying unary negative.

Reply:
read_bytes is called (initially) from `calculate_heap_usage`, and in that case there is a check on `num_bytes` (extracted from the encoded term) to ensure it is no larger than 8.  It is called again in `parse_external_terms`, but we have a MAINTENANCE NOTE in the code that states that `parse_external_terms` is always preceded by a call to `calculate_heap_usage`, in order to ensure that heap allocations won’t require a GC.

So technically, in this static code, `num_bytes` will never be greater than 8 (and hence `i` will be less than 8.

Do you think we still need a redundant check here?  I have been going with the contract implicit in the maintenance note, but adding a check is no big deal.